src/RequestParser.php

```diff
@@ -118,4 +85,38 @@ protected function inlineFiles(Request $request): array
 
         return $operations;
     }
+
+    /**
+     * Extracts the body parameters from the request.
+     *
+     * @return array<mixed>
+     */
+    protected function bodyParams(Request $request): array
+    {
+        /**
+         * Never null, since Symfony defaults to application/x-www-form-urlencoded.
+         *
+         * @var string $contentType
+         */
+        $contentType = $request->header('Content-Type');
+
+        if (Str::startsWith($contentType, 'multipart/form-data')) {
+            return $this->inlineFiles($request);
+        }
+
+        $bodyParams = $request->input();
+
+        if (is_array($bodyParams) && Arr::isAssoc($bodyParams)) {
+            return $bodyParams;
+        }
+
+        if (Str::startsWith($contentType, 'application/graphql')) {
+            /** @var string $content */
+            $content = $request->getContent();
+
+            return ['query' => $content];
+        }
+
+        throw new BadRequestGraphQLException('Could not parse GraphQL request body');
```

Comment:
Can we give the user a better indication as to why we could not parse it?

Reply:
Since we don't do anymore the json_decode ourselves, this became probably out of scope. This could only occur on malformed payload (eg. sending json data with a application/x-www-form-urlencoded header).
I'm not sure on how to improve this message here. Would be happy if you have any suggestion .

Reply:
We might include the content-type header at least.